apis/workstations/v1alpha1/config_reference.go

```diff
@@ -148,6 +150,8 @@ func NewWorkstationConfigRef(ctx context.Context, reader client.Reader, obj *Wor
 		return nil, fmt.Errorf("cannot reset `metadata.name` or `spec.resourceID` to %s, since it has already assigned to %s",
 			resourceID, actualResourceID)
 	}
+	id.Name = resourceID
+	id.Namespace = obj.GetNamespace()
```

Comment:
They are used in the `NormalizedExternal` function. Ref: https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/bc85425a3f65f771296e8406702000b2056ba758/apis/workstations/v1alpha1/config_reference.go#L49

That function is currently unused, but it will probably be used soon by `WorkstationsWorkstation`.

Reply:
This might be a better example on how the `NewWorkstationConfigRef` and `NormalizedExternal` could be used. 
[NormalizedExternal
](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/apis/secretmanager/v1beta1/secret_reference.go)
https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/apis/secretmanager/v1beta1/secret_identity.go#L52

In summary, they should be in different struct (and we have fixed that in the api template), and when using the `WorkstationConfig` as a reference, the `name` or `external` is assigned from the API, and we call `normalizeExternal`.

Reply:
Ok, updated to use the latest revision of the ref resolving template. Also, modified the KRM type definition to remove the unnecessary `location` and `projectRef` fields.